src/hooks/useAsyncError.ts

```diff
@@ -0,0 +1,16 @@
+import { useState, useCallback } from "react";
+
+const useAsyncError = () => {
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [_, setError] = useState();
+
+  const asyncErrorHandler = useCallback((e: Error) => {
+    setError(() => {
+      throw e;
+    });
+  }, [setError]);
+
+  return { asyncErrorHandler };
+};
```

Comment:
Not sure i  understand the value of this function. Do we need to `useState`, why can't the calling code just throw the error?

Reply:
Because `react-error-boundary` doesn't catch async errors. Therefore, we need to make them synchronous.